tests/Microsoft.Bot.Streaming.Tests/NamedPipeTests.cs

```diff
@@ -28,21 +27,32 @@ public void Close_DisconnectsStreamAsync()
 
             try
             {
+                // The ConnectAsync task returns only after the connection has been disposed.
+                // In this context it makes for ugly code, but in the context of blocking an HTTP
+                // response until the the streaming connection has ended it creates an easier to
+                // follow user experience.
                 reader.ConnectAsync();
-                writer.StartAsync();
+                await writer.StartAsync();
 
-                readStream.WaitForConnectionAsync().ConfigureAwait(false);
-                writeStream.ConnectAsync(500).ConfigureAwait(false);
-                writer.Disconnect();
+                // The writeStream can only connect to the readStream if the readStream is listening for new connections.
+                // This creates a dependency between the two tasks below, requiring the two to be running in parallel until
+                // the readStream receives an incoming connection and the writeStream establishes an outgoing connection.
+                Task.WaitAll(new Task[] { readStream.WaitForConnectionAsync(), writeStream.ConnectAsync() });
```

Comment:
>Task.WaitAll [](start = 16, length = 12)

This should be using WhenAll which returns a task which completes when all of the tasks inside complete:
    await Task.WhenAll(....)

Also, WhenAll and WaitAll() have params attribute, which means no need for the new Task[] { ... } just do
    await Task.WhenAll(task1, task2);

Reply:
Thank you for the context! I've updated to use WhenAll.